library/spdm_transport_mctp_lib/libspdm_mctp_common.c

```diff
@@ -22,7 +22,9 @@
  * @retval RETURN_SUCCESS               The message is encoded successfully.
  * @retval RETURN_INVALID_PARAMETER     The message is NULL or the message_size is zero.
  **/
-libspdm_return_t libspdm_mctp_encode_message(const uint32_t *session_id, size_t message_size,
+libspdm_return_t libspdm_mctp_encode_message(const uint32_t *session_id,
```

Comment:
Production code should (hopefully) not have to be changed to fix this. Why can't this be restricted to `unit_test/`? In particular why can't `LIBSPDM_TEST_ALIGNMENT` be `1`?

Reply:
MCTP is NOT impacted, because the LIBSPDM_MCTP_ALIGNMENT is always 1 at this moment.
MCTP is changed, just because I want to align the code with TEST_ALIGNMENT.


TEST is impacted, because I do want to see if our code can handle 4 alignment correctly.

Reply:
I am now slightly concerned about libspdm checking the exact size of an SPDM message. Is it possible that there exist transports such that 277's `Application Data Length` may not accurately give the exact size of the SPDM message?

Reply:
That might be a question for the SPDM Working Group.

Reply:
Does exact size check block this PR?

If no, I suggest we file a new issue.

Reply:
Ok that can be a separate issue.